### ISSUE TITLE
error:  failed to update card

### ISSUE THREAD

**Issue by iKrushYou** [uhppoted/uhppoted-httpd#27]
Hey, currently getting an error when trying to create a card from the user interface

```
uhppoted-httpd  | 2024/04/23 00:58:29 WARN   423137505  failed to update card 1234567
```

I went to cards, clicked +, added a user name, card number, end date, and hit check. then got this in the console. I haven't been able to find a working combination (I've tried others). I can confirm that users created in the stock web UI and the CLI show up eventually. 

as a comparison, I set up the `uhppoted-cli` as well and was able to successfully add a card that way. 

running using docker compose, with image
```
ghcr.io/uhppoted/httpd   latest    e47723788c1c   3 weeks ago   30.3MB
```

What kind of information would be helpful to provide?

**Comment by uhppoted** [comment]
I've been able to reproduce this on my own system - let me look into it and get back to you..

**Comment by iKrushYou** [comment]
Thanks for such prompt responses! Do you happen to have a discord server or a less formal place to chat? I have minor questions but don't want to keep raising GitHub issues

**Comment by uhppoted** [comment]
Ah - got it! You need to provide both a _card start date_ and a _card end date_ (_From_ and _To_) for the controller to accept a card.

The UI doesn't require a card start and end date because you might be e.g. preloading a set of cards which will get assigned to actual people for actual use later - it's only showing up as a problem now in _uhppoted-httpd_ because of a recent system-wide change to make card date/time handling more robust.

I'm going to have to think about this one a bit because automatically assigning a default card start and end date is potentially problematic (e.g. a temporary access card may be unexpectedly valid) - but hopefully the workaround will get you up and running for now.

re. less formal place to chat. Most people seem to just use the overarching [uhppote discussions](https://github.com/uhppoted/uhppoted/discussions)  area - which is quite nice because it created a centralised, searchable repository for ad hoc information and ideas. [Discussions](https://github.com/uhppoted/uhppoted-httpd/discussions) are enabled for this repo too if you prefer something more topic specific.

**Comment by iKrushYou** [comment]
thank you! I found the start / end date requirements interesting. I would want all of the cards I am entering to have an infinite or indeterminate end date. Especially since there doesn't seem to be a bulk way to extend card end dates. We're going to have a couple hundred cards saved into the system that will all be active "indefinitely." I could set the end date myself to something very far out, but also I wouldn't want to rely on future admins to have to know what to set the end date to

I'd likely want to make a change to create a default "MAX" end date

**Comment by twystd** [comment]
Ref. [Manage cards without valid start/end dates](https://github.com/uhppoted/uhppoted-httpd/issues/29)

FYI, have added support for (optional) default card start and end dates (and also ignored cards with missing start or end dates when downloading cards to the controllers. It's only available in the _development_ version for the moment so you would need to clone and build it - but it will be released along with v0.8.9 which should hopefully be within the next week or so..

**Comment by iKrushYou** [comment]
Thanks, I appreciate the update!

**Comment by twystd** [comment]
Closing as fixed by [#29](https://github.com/uhppoted/uhppoted-httpd/issues/29), pending release v0.8.9
